### PR TITLE
Fix React component Flow types

### PR DESCRIPTION
**Pull request by tlrobinson**
In a lot of places we were doing:

```javascript
class Foo extends Component<*, Props, State> {
```

instead of

```javascript
class Foo extends Component {
  props: Props;
  state: State;
```

so a lot of the types we wrote up weren't being type checked.

### CODE REVIEW

**Comment by tlrobinson** [comment]
@metabase/core-developers could someone review this please?

File: frontend/src/metabase/public/components/widgets/EmbedModalContent.jsx [hunk 1]
```diff
@@ -14,36 +14,36 @@ import { getSignedPreviewUrl, getUnsignedPreviewUrl, getSignedToken } from "meta
...
-    secretKey: string,
     resource: EmbeddableResource,
     resourceType: string,
     resourceParameters: Parameter[],
+
+    isAdmin: boolean,
+    siteUrl: string,
+    secretKey: string,
+
+    // Flow doesn't understand these are provided by @connect?
```
Comment (attekei) [inline frontend/src/metabase/public/components/widgets/EmbedModalContent.jsx]: Seems that Flow doesn't understand object decorators. By default it warns about them, but we have suppressed the warnings in `.flowconfig`. So probably the right way to fix this is to use the old-school `export default connect(...)(...);` instead.

Comment (attekei) [inline frontend/src/metabase/public/components/widgets/EmbedModalContent.jsx]: (but I don't consider this as a blocking issue)

**Review by attekei — COMMENTED** [review]
Generally looks good! Two issues that are not blockers but which are good to discuss.

File: frontend/src/metabase/public/components/EmbedFrame.jsx [hunk 2]
```diff
@@ -32,9 +32,14 @@ type Props = {
     setParameterValue: (id: string, value: string) => void
 }
 
+type State = {
```
Comment (tlrobinson) [inline frontend/src/metabase/public/components/EmbedFrame.jsx]: I like to always explicitly declare them up top as a sort of documentation.

Comment (attekei) [inline frontend/src/metabase/public/components/EmbedFrame.jsx]: 👍  I will start doing that as well

**Review by kdoh — APPROVED** [review]

**Pull request merged**